elsa-types.el

```diff
@@ -476,5 +476,101 @@ other, then this is a supertype of other."
 (cl-defmethod elsa-type-describe ((this elsa-generic-type))
   (symbol-name (oref this label)))
 
+;; One-dimensional sparse arrays indexed by characters
+(defclass elsa-type-chartable (elsa-type) ())
+
+(cl-defmethod elsa-type-describe ((this elsa-type-chartable))
+  "Char-table")
+
+;; One-dimensional arrays of t or nil.
+(defclass elsa-type-boolvector (elsa-type) ())
+
+(cl-defmethod elsa-type-describe ((this elsa-type-boolvector))
+  "Bool-vector")
+
+;; Super-fast lookup tables
+(defclass elsa-type-hashtable (elsa-type) ())
+
+(cl-defmethod elsa-type-describe ((this elsa-type-hashtable))
+  "Hash table")
+
+;; Compound objects with programmer-defined types
+(defclass elsa-type-record (elsa-type) ())
```

Comment:
I'm not sure about this one.  As I understand it this is implementation detail for `cl-defstruct`.  Maybe it will make no sense once we actually implement class types and struct types.  We can remove it later I guess because I really don't know how it would be useful.